### PR TITLE
[FE] [FIX] fix edit category

### PR DESCRIPTION
**Pull request by wernerramon**
### 🛠 Changes being made

when editing a category the dishes from the old category get mapped to the new category
### ✨ What's the context?

What's the context for the changes? Are there any

### 🧠 Rationale behind the change

Why did you choose to make these changes? Were there any trade-offs you had to consider?

### 🧪 Test plan

How do you know the changes are safe to ship to production?

### 📸 Screenshots (optional)

If you made UI changes, what are the before an afters?

### 🏎 PR checklist

1. Read the ticket
- [ ] Read the related ticket on Jira and understand the requirements for this ticket. In case anything is unclear, ask the ticket creator or ticket executor.
2. Matching PR description
- [ ] Check that the PR description on GitHub is aligned with the ticket description on Jira.
3. Review the changes on GitHub
- [ ] is the current version of the destination branch (mostly main) merged into the PR branch?
- [ ] does it fulfill the definition of done for the ticket?
- [ ] were the tests adjusted or extended for the changed behavior?
- [ ] was the documentation adjusted or extended for changed behavior?
- [ ] does it align with the team code guidelines?
- [ ] check readability of code - is it easy to understand what is happening in the code snippets?
- [ ] proper naming
- [ ] is it logical?
- [ ] are there any performance issues?
- [ ] could it be improved / simplified?
4. Execute locally / in testing environment
- [ ] Execute the code locally and check if there are any issues. Test whether the new behavior from the ticket acts as expected. Try to think of edge cases and test them.
5. Run tests
- [ ] Run the tests locally and check that all of them succeed.
6. Linting & Formatting
- [ ] Check that the changes didn’t cause any linting or formatting warnings of errors.
7. Submit Feedback
- [ ] Approve if all of the following points fulfilled:
   - all previous steps did not reveal any issues
   - there are no open comments on the PR

- [ ] otherwise, request changes with specific comments (constructive) on what to change!

### CODE REVIEW

File: packages/backend/src/controllers/restaurantController.ts [hunk 1]
```diff
@@ -710,6 +710,24 @@ export async function addCategory(
     }));
     rest.mealType = transformedArray;
 
+    rest.dishes.forEach(dish => {
```
Comment (JosefineM2000) [inline packages/backend/src/controllers/restaurantController.ts:713]: Rather:

`const editedCategory = newCategories.find(category => category.edited);

      if (editedCategory) {
          rest.dishes.forEach(...)
      }`

File: packages/backend/src/controllers/restaurantController.ts [hunk 2]
```diff
@@ -710,6 +710,24 @@ export async function addCategory(
...
+      const dishCategory = dish.category;
+    
+      const editedCategory = newCategories.find(category => category.edited);
+    
+      if (editedCategory) {
+        const categoryInNewCategories = newCategories.some(category =>
```
Comment (JosefineM2000) [inline packages/backend/src/controllers/restaurantController.ts:719]: Did you mean editedCategories.some ?

File: packages/backend/src/controllers/restaurantController.ts [hunk 3]
```diff
@@ -733,6 +751,8 @@ export async function addCategory(
       menuDesignID: rest.menuDesignID as number,
     });
 
+    console.log('restaurantBE dishes: ', restaurantBE.dishes);
```
Comment (JosefineM2000) [inline packages/backend/src/controllers/restaurantController.ts:754]: please remove again

File: packages/backend/src/controllers/restaurantController.ts
Comment (JosefineM2000) [inline packages/backend/src/controllers/restaurantController.ts:1]: I just tested and I think we need to add a new category with the editedName in all Restos the dish is assigned to where this category name doesnt exist. Otherwise the Dish edit form only shows one restaurant, even if the dish was assigned to several restos before.

File: packages/backend/src/controllers/restaurantController.ts [hunk 4]
```diff
@@ -825,3 +758,87 @@ export async function addCustomerResto(userID: number, customerID: string) {
...
+    await rest.save();
+
+    if (editedCategory) {
+      const restos = await Restaurant.find({ userID: userID });
+
+      for (const resto of restos) {
+        if (resto._id !== uid) {
+          const lastElement = resto.mealType[resto.mealType.length - 1];
+          const newId = resto.mealType.length;
+          const newSortId = lastElement ? lastElement.sortId as number + 1 : 1;
```
Comment (JosefineM2000) [inline packages/backend/src/controllers/restaurantController.ts:809]: doesnt necessarily work, as the user can define sortIDs as [1, 3, 4] (2 missing). Rather get a list of all sortIDs and use max()

File: packages/backend/src/controllers/restaurantController.ts [hunk 5]
```diff
@@ -825,3 +758,87 @@ export async function addCustomerResto(userID: number, customerID: string) {
...
+        }
+      }
+    });
+    await rest.save();
+
+    if (editedCategory) {
+      const restos = await Restaurant.find({ userID: userID });
+
+      for (const resto of restos) {
+        if (resto._id !== uid) {
```
Comment (JosefineM2000) [inline packages/backend/src/controllers/restaurantController.ts:806]: if resto._id !== uid && resto includes the dish that is affected - you can check by dishName as its user specific

File: packages/backend/src/controllers/restaurantController.ts
Comment (JosefineM2000) [inline packages/backend/src/controllers/restaurantController.ts:1]: In the restos with affected dishes (not the ones where category is edited, but where category is added due to assigned dish), the dish is displayed twice in the menu. check menu BE method to find root cause, please

**Review by JosefineM2000 — APPROVED** [review]
